api/apps/v1alpha1/nemo_guardrails_types.go

```diff
@@ -57,8 +57,8 @@ type NemoGuardrailSpec struct {
 	Command []string        `json:"command,omitempty"`
 	Args    []string        `json:"args,omitempty"`
 	Env     []corev1.EnvVar `json:"env,omitempty"`
-	// The name of an secret that contains authn for the NGC NIM service API
-	AuthSecret string `json:"authSecret"`
+	// The name of an secret that contains authn for the NGC NIM service API, required when the NIM is hosted by NGC
+	AuthSecret *string `json:"authSecret,omitempty"`
```

Comment:
Pointer type is not required here. Pointer type is useful when we need to distinguish between unset (nil) and empty values (""), this is not the case here. Empty value have no special meaning in this case.